File: social-map/src/main/java/com/mapsocial/service/ShopService.java
package com.mapsocial.service;

import com.mapsocial.dto.request.shop.CreateShopRequest;
import com.mapsocial.dto.request.shop.UpdateShopRequest;
import com.mapsocial.dto.response.shop.ShopResponse;

import java.util.List;
import java.util.UUID;

public interface ShopService {
    ShopResponse createShop(UUID userId, CreateShopRequest request);
    ShopResponse updateShop(UUID userId, UUID shopId, UpdateShopRequest request);
    void deleteShop(UUID userId, UUID shopId);
    ShopResponse getShopById(UUID shopId);
    List<ShopResponse> getAllShops();
}

File: social-map-fe/src/services/shopService.js
import { api } from "./apiClient";

/**
 * Tạo mới cửa hàng
 * @param {Object} shopData
 * @returns {Promise<Object>} ShopResponse
 */
export const createShop = async (shopData) => {
  return await api.post("/shops", shopData);
};

/**
 * Cập nhật cửa hàng
 * @param {string} shopId - UUID của shop
 * @param {Object} shopData
 * @returns {Promise<Object>} ShopResponse
 */
export const updateShop = async (shopId, shopData) => {
  return await api.put(`/shops/${shopId}`, shopData);
};

/**
 * Xóa cửa hàng
 * @param {string} shopId - UUID của shop
 * @returns {Promise<void>}
 */
export const deleteShop = async (shopId) => {
  return await api.delete(`/shops/${shopId}`);
};

/**
 * Lấy cửa hàng theo ID
 * @param {string} shopId - UUID của shop
 * @returns {Promise<Object>} ShopResponse
 */
export const getShopById = async (shopId) => {
  return await api.get(`/shops/${shopId}`);
};

/**
 * Lấy tất cả cửa hàng
 * @returns {Promise<Array>} Danh sách ShopResponse
 */
export const getAllShops = async () => {
  return await api.get("/shops");
};
